service/src/main/java/com/epam/esm/service/BaseService.java

```diff
@@ -0,0 +1,16 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dto.GiftCertificateDTO;
+import com.epam.esm.exception.ServiceException;
+
+import java.util.List;
+
+public interface BaseService<T> {
+
+    List<T> findAll() throws ServiceException;
+    T find(String name) throws ServiceException;
+    void create(T bean) throws ServiceException;
```

Comment:
Not sure that create operation should be void. I think you need to return result model on client.